services/kvstore/src/main/java/org/hyperledger/besu/services/kvstore/SegmentedKeyValueStorageTransactionValidatorDecorator.java

```diff
@@ -43,6 +44,14 @@ public SegmentedKeyValueStorageTransactionValidatorDecorator(
     this.transaction = toDecorate;
   }
 
+  @Override
+  public Optional<byte[]> get(final SegmentIdentifier segmentId, final byte[] key)
+      throws StorageException {
+    checkState(active, "Cannot invoke get() on a completed transaction.");
```

Comment:
Why can't we invoke get on a completed transaction?

Reply:
For a get maybe it is fine? I was just following the pattern that the other methods did which was not operations on a completed transaction. Hopefully, a storage expert will know the answer.

Reply:
I think it makes sense to throw in this case cause we want to avoid returning values we are uncertain if it's going to be committed or not. Think chances are minimal but there is a scenario where this could happen. @garyschulte what you reckon?